### PR TITLE
[Wave Collect] [Xero] More features text

### PR DESCRIPTION
**Pull request by mananjadhav**
### Fixed Issues

$ https://github.com/Expensify/App/issues/42644
PROPOSAL:

### Tests

**Tests**

**Connected Integration**
1. Open any workspace with an existing accounting integration - say, `Xero` or QBO..
2. Click on `More Features`.
3. Go to `Organize` section and try clicking on any of the switches.
4. Verify that you see a modal with the following content.

image

5. Verify that you see the connected integration name in the title.
6. Verify that clicking cancel closes the modal.
7. Verify that click `Manage settings` takes you to the Accounting page.
8. Repeat the steps 3-7 for `Integrate` section and the following modal shows up.

image

9. Repeat all the steps with the Spanish language.

**Not Connected Integration**
1. Open a workspace that doesn't have the any accounting integration.
2. Click on `More Features`.
3. Go to organize section and try clicking on any of the switches.
4. Verity that you're able to toggle on and off, and accordingly able to see Categories, Taxes, Tags pages.

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

Same as QA Steps.

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

**With new copy**

https://github.com/Expensify/App/assets/3069065/31fb62d6-89bf-4b57-bc47-2cd71c3c6be8

**Old Video**

https://github.com/Expensify/App/assets/3069065/fdc6a30f-668b-4fa9-b6bc-794c0dc02985

</details>

<details>
<summary>Android: mWeb Chrome</summary>

**With new copy**

https://github.com/Expensify/App/assets/3069065/9d654106-c041-4ac2-8a69-1ba32a1e7a06

**Old Video**

https://github.com/Expensify/App/assets/3069065/92329802-7757-4401-999b-851ccacabc02

</details>

<details>
<summary>iOS: Native</summary>

**With new copy**

https://github.com/Expensify/App/assets/3069065/1741c2c6-a8b3-463a-b6a2-6ee9d15b2195

**Old Video**

https://github.com/Expensify/App/assets/3069065/ba5afadd-a639-4c1b-8fd3-cc33d8c3dcf2

</details>

<details>
<summary>iOS: mWeb Safari</summary>

**With new copy**

https://github.com/Expensify/App/assets/3069065/88b2d072-9b4c-438b-85ab-cbfee0c94d02

**Spanish Translations**

mweb-safari-connection-modal-es-1

mweb-safari-connection-modal-es-2

**Old Video**

https://github.com/Expensify/App/assets/3069065/644224b3-4b05-4ad8-9f54-6d3f2dbf5658

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

**With new copy**

https://github.com/Expensify/App/assets/3069065/2a8424c6-c723-426d-8cae-7609df28e851

**New Spanish Translations**

web-connection-modal-es-1

web-connection-modal-es-2

**Modal with off switch state**

https://github.com/Expensify/App/assets/3069065/27233083-2e3b-4a45-94f6-11f362aa0ac3

**Old Video**
https://github.com/Expensify/App/assets/3069065/29b5a8b1-7851-4ed7-8a18-d5cec097d366

https://github.com/Expensify/App/assets/3069065/c6d25bc2-3d67-4254-a293-7bfab83012e9

</details>

<details>
<summary>MacOS: Desktop</summary>

**With new copy**

https://github.com/Expensify/App/assets/3069065/f65f389a-ea3d-44a8-ba13-79bfd7733e79

**Old Video**

https://github.com/Expensify/App/assets/3069065/9997d434-1295-4b0f-97cd-5e6a4604a278

</details>

### CODE REVIEW

**Comment by mananjadhav** [comment]
https://github.com/Expensify/App/assets/3069065/f69d9652-6f2a-4783-840a-899b0fa1ef1c

**Comment by dubielzyk-expensify** [comment]
This is looking so good so far, but I'll await final review for when the PR is ready 😄

File: src/pages/workspace/WorkspaceMoreFeaturesPage.tsx
Comment (hungvu193) [inline src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:110]: We repeated this piece of code few times, I think we should reuse it with an `useCallback`?

Comment (mananjadhav) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:110]: I am not sure if it is necessary, we're not relying on the previous state that we really need to put it inside useCallback. The `hasAccountingConnection` is also dependent on the prop.

And about repetition, it's a setState call. I could wrap it inside the method but looks like an overkill.

Comment (hungvu193) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:110]: I think we should, including ` Policy.enablePolicyCategories(policy?.id ?? '', isEnabled);` line below.
Also can you complete the checklist please 😄 ?

Comment (mananjadhav) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:110]: Working through the checklist.

**Comment by mananjadhav** [comment]
@rojiphil Please ignore this PR. @hungvu193 is already reviewing it.

**Comment by mananjadhav** [comment]
@Expensify/design For your review.

@lakchote @pecanoro if you folks can review the Spanish translations.

File: src/pages/workspace/WorkspaceMoreFeaturesPage.tsx
Comment (mananjadhav) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:110]: With enablePolicyCategories it might make sense. Let me check.

**Comment by hungvu193** [comment]
@mananjadhav There will be an edge case where the modal is showing but we disconnect our Xero (or QBO) from other devices. Do you think it's worth fixing?

**Comment by mananjadhav** [comment]
> @mananjadhav There will be an edge case where the modal is showing but we disconnect our Xero (or QBO) from other devices. Do you think it's worth fixing?

I don't think so. This behavior exists in all pages. Take Delete workspace for example.

File: src/pages/workspace/WorkspaceMoreFeaturesPage.tsx
Comment (mananjadhav) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:110]: @lakchote Do you think it makes sense to move this code inside a useCallback?

**Comment by shawnborton** [comment]
Looking good to me!

**Comment by hungvu193** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/Expensify/App/assets/16502320/0883983e-d239-46eb-8990-7daa19db0638


</details>

<details>
<summary>Android: mWeb Chrome</summary>



https://github.com/Expensify/App/assets/16502320/7f857854-ec83-42cc-afd8-2d3bbf5ac39d


</details>

<details>
<summary>iOS: Native</summary>



https://github.com/Expensify/App/assets/16502320/ee9ab444-2542-4af2-9023-0f6a2ef4cab0


</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/Expensify/App/assets/16502320/02585ccb-ab7e-4f12-8987-8046641a1dac


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/Expensify/App/assets/16502320/3d478073-84f0-46d4-bd96-1f7304a9861c


</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/Expensify/App/assets/16502320/d74bead6-f061-4c68-8fa5-3db3fa9dfb51


</details>

**Comment by trjExpensify** [comment]
Are we running with that copy? I thought we landed [on this](https://expensify.slack.com/archives/C036QM0SLJK/p1716548406024769?thread_ts=1716411028.782029&cid=C036QM0SLJK) in the end: 

image

File: src/pages/workspace/WorkspaceMoreFeaturesPage.tsx
Comment (lakchote) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:110]: I do agree with @mananjadhav, I don't think there's a benefit _here_ to use `useCallback()`. It's not a computational intensive operation, we're just setting state. It can be tempting to use it to avoid repetition, but using these memorized functions come at a cost on intial render. Given the fact it's just setting state, I'd say it's a premature optimization scenario and we can keep it that way.

Happy to be proven wrong though, if you can give me arguments proving me otherwise @hungvu193 😄

**Comment by mananjadhav** [comment]
Ohh I wasn't aware. I am afk right now. But I can update the copy. I relied on the GH comment [here](https://github.com/Expensify/App/pull/42487#issuecomment-2127707089). 

So the ones attached are the final ones?

**Comment by trjExpensify** [comment]
Can you drop these in and then I'm tagging @Gonals @pecanoro for a second opinion! 

**Not so fast...**
To enable or disable this feature change your accounting import settings. 
[Manage settings]
[Cancel]

**No tan rápido...**
Para activar o desactivar esta función, cambie la configuración de importación contable.
[Gestionar la configuración]
[Cancelar]

__________
**Not so fast...**
To enable or disable this feature change your accounting import settings. 
[Manage settings]
[Cancel]

**No tan rápido...**
Desconecte su conexión contable del espacio de trabajo si desea desactivar la Contabilidad.
[Gestionar la configuración]
[Cancelar]

**Comment by mananjadhav** [comment]
I've updated the videos with the new copy. For Spanish I've only posted for Web and mWeb Safari.

File: src/pages/workspace/WorkspaceMoreFeaturesPage.tsx
Comment (hungvu193) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:107]: I think that's fine though. We can keep it this way :)

**Review by hungvu193 — APPROVED** [review]
LGTM 🚀 
All yours @lakchote

**Review by lakchote — APPROVED** [review]

**Comment by Gonals** [comment]
I believe our standard is to not treat the user as "usted", so a slight change:

**Not so fast...**
To enable or disable this feature change your accounting import settings. 
[Manage settings]
[Cancel]

**No tan rápido...**
Para activar o desactivar esta función, cambia la configuración de importación contable.
[Gestionar la configuración]
[Cancelar]

__________
**Not so fast...**
To enable or disable this feature change your accounting import settings. 
[Manage settings]
[Cancel]

**No tan rápido...**
Desconecta tu conexión contable del espacio de trabajo si deseas desactivar la contabilidad.
[Gestionar la configuración]
[Cancelar]


Plus, is the second English one wrong? It seems like a duplicate of the first one 😕

**Comment by mananjadhav** [comment]
@Gonals  Done.

Screenshot 2024-05-28 at 7 37 31 PM

**Review by Gonals — APPROVED** [review]

**Pull request merged**
